@here/harp-geometry/test/ClipPolygonTest.ts

```diff
@@ -178,7 +179,7 @@ describe("ClipPolygon", () => {
         assert.isTrue(polygon.some(vert => vert.y < 0));
         assert.isTrue(polygon.some(vert => vert.y > extents));
 
-        const clippedPolygon = clipPolygon(polygon, tileBounds);
+        const clippedPolygon = clipPolygon(polygon, extents);
 
         assert.isNotEmpty(clippedPolygon);
 
```

Comment:
I would add a test with concave polygon resulting into multiple parts after clipping.

```
    it("Concave polygon resulting into 2 parts after clipping", () => {
        const polygon: Vector2[] = [
            new Vector2(-100, 0),
            new Vector2(4096, 0),
            new Vector2(-50, 2048),
            new Vector2(4096, 4096),
            new Vector2(-100, 4096)
        ];

        const expectedPolygon: Vector2[] = [
            new Vector2(0, 0),
            new Vector2(4096, 0),
            new Vector2(0, 2023),
            new Vector2(0, 2073),
            new Vector2(4096, 4096),
            new Vector2(0, 4096)
        ];

        const clippedPolygon = clipPolygon(polygon, extents);
        assert.strictEqual(JSON.stringify(clippedPolygon), JSON.stringify(expectedPolygon));
    });
```

Reply:
done, after clipping it results in two triangles.